色花堂没提供magnet连结的文章无法下载

### 路由地址
https://rsshub.app/dsndsht23/103/480

### 预期是什么？
有码高清的全部文章都能透过Flexget或者Synology的Download Station下载。

### 实际发生了什么？
没提供magnet连结的文章无法下载。

### 额外信息（日志、报错等）
---2021/1/17---
查看色花堂的路由得知抓的是以下内容，
const enclosureUrl = postMessage.find('div.blockcode li').text();
但特定发布者的文章blockcode无资料，或者blockcode的内容为片片资讯，以至导出的enclosureUrl错误。

因此是否能改为优先侦测blockcode的内容是否带有magnet连结，若无则改提供附件的.torrent。

或者enclossureUrl直接抓取附件的.torrent。

@junfengP 大佬，能请您协助查看下吗？谢谢。


---2021/1/15---
[这篇文章](https://www.sehuatang.net/forum.php?mod=viewthread&tid=443297&extra=page%3D1%26filter%3Dtypeid%26typeid%3D480)导出的下载连结是
https://www.sehuatang.net/forum.php?mod=viewthread&tid=443297&extra=page%3D1%26filter%3Dtypeid%26typeid%3D480
以至下载的会是forum.php。

[这篇文章](https://www.sehuatang.net/forum.php?mod=viewthread&tid=442559&extra=page%3D2%26filter%3Dtypeid%26typeid%3D480)导出的下载连结是
General Unique ID                      : 71335113643678085749467464322699868043(...后略)
会跳错误，无法下载。

做了如下改动，因为手边没有开发环境 所以测试不了
```
--- a/lib/routes/dsndsht23/index.js
+++ b/lib/routes/dsndsht23/index.js
@@ -1,5 +1,6 @@
 const got = require('@/utils/got');
 const cheerio = require('cheerio');
+const url = require('url');
 
 const host = 'https://www.sehuatang.net/';
 
@@ -90,15 +91,17 @@ module.exports = async (ctx) => {
                 }
             }
             const description = (postMessage.html() || '抓取原帖失败').replace(/ignore_js_op/g, 'div');
-            const enclosureUrl = postMessage.find('div.blockcode li').text();
 
+            const magnet = postMessage.find('div.blockcode li').text();
+            const isMag = magnet.startsWith("magnet");
+            const enclosureUrl = isMag? magnet : url.resolve(host, postMessage.find("a[href^=forum\.php\?mod\=attachment]:not([href$=nothumb\=yes])").attr("href"));
             const single = {
                 title: title,
                 link: itemUrl,
                 description: description,
                 pubDate: new Date(date).toUTCString(),
                 enclosure_url: enclosureUrl,
-                enclosure_type: 'application/x-bittorrent',
+                enclosure_type: isMag? 'application/x-bittorrent': 'application/octet-stream',
             };
             if (description !== '抓取原帖失败') {
                 ctx.cache.set(itemUrl, JSON.stringify(single));
```

执行`npm run format`提示如下：
```
/root/RSSHub/lib/routes/dsndsht23/index.js
  97:100  error  Unnecessary escape character: \.  no-useless-escape
  97:105  error  Unnecessary escape character: \?  no-useless-escape
  97:110  error  Unnecessary escape character: \=  no-useless-escape
  97:142  error  Unnecessary escape character: \=  no-useless-escape
```
> 因为我是在`chrome`的控制台测试的CSS选择器，所以需要对这些字符转义，不知道在nodejs中不转义是否正常

`application/octet-stream`是从`torrent`文件下载的响应头中获取的，不知道Flexget或者Synology的Download Station能否解析

--- a/lib/routes/dsndsht23/index.js
+++ b/lib/routes/dsndsht23/index.js
@@ -1,5 +1,6 @@
 const got = require('@/utils/got');
 const cheerio = require('cheerio');
+const url = require('url');
 
 const host = 'https://www.sehuatang.net/';
 
@@ -90,7 +91,10 @@ module.exports = async (ctx) => {
                 }
             }
             const description = (postMessage.html() || '抓取原帖失败').replace(/ignore_js_op/g, 'div');
-            const enclosureUrl = postMessage.find('div.blockcode li').text();
+
+            const magnet = postMessage.find('div.blockcode li').text();
+            const isMag = magnet.startsWith('magnet');
+            const enclosureUrl = isMag ? magnet : url.resolve(host, postMessage.find('a[href^=forum\\.php\\?mod\\=attachment]:not([href$=nothumb\\=yes])').attr('href'));
 
             const single = {
                 title: title,
@@ -98,7 +102,7 @@ module.exports = async (ctx) => {
                 description: description,
                 pubDate: new Date(date).toUTCString(),
                 enclosure_url: enclosureUrl,
-                enclosure_type: 'application/x-bittorrent',
+                enclosure_type: isMag ? 'application/x-bittorrent' : 'application/octet-stream',
             };
             if (description !== '抓取原帖失败') {
                 ctx.cache.set(itemUrl, JSON.stringify(single));

